chalicelib/checks/wrangler_checks.py

```diff
@@ -472,3 +472,172 @@ def patch_file_size(connection, **kwargs):
+            # append if cell_line field is missing
+            if not biosource.get('cell_line'):
+                missing.append({'uuid': biosource['uuid'],
+                                '@id': biosource['@id'],
+                                'biosource_type': biosource.get('biosource_type'),
+                                'description': biosource.get('description'),
+                                'error': 'Missing cell_line metadata'})
+                # detail = 'In Biosource {}'.format(value['@id']) + \
+                #      ' - Missing Required cell_line field value for biosource type  ' + \
+                #      '{}'.format(bstype)
+    check.full_output = missing
+    check.brief_output = [item['uuid'] for item in missing]
+    if missing:
+        check.status = 'WARN'
+        check.summary = 'Cell line biosources found missing cell_line metadata'
+    else:
+        check.status = 'PASS'
+        check.summary = 'No cell line biosources are missing cell_line metadata'
+    return check
+
+
+@check_function()
+def external_expsets_without_pub(connection, **kwargs):
+    '''
+    checks external experiment sets to see if they are attributed to a publication
+    '''
+    check = init_check_res(connection, 'external_expsets_without_pub')
+
+    ext = ff_utils.search_metadata('search/?award.project=External&type=ExperimentSet',
+                                   ff_env=connection.ff_env, page_limit=200)
+    no_pub = []
+    for expset in ext:
+        if not expset.get('publications_of_set') and not expset.get('produced_in_pub'):
+            # add to Output
+            no_pub.append({'uuid': expset['uuid'],
+                           '@id': expset['@id'],
+                           'description': expset.get('description'),
+                           'lab': expset['lab'].get('display_title'),
+                           'error': 'Missing attribution to a publication'})
+    if no_pub:
+        check.status = 'WARN'
+        check.summary = 'External experiment sets found without associated publication'
+        check.description = '{} external experiment sets are missing attribution to a publication.'.format(len(no_pub))
+    else:
+        check.status = 'PASS'
+        check.summary = 'No external experiment sets are missing publication'
+        check.description = '0 external experiment sets are missing attribution to a publication.'
+    check.full_output = no_pub
+    check.brief_output = [item['uuid'] for item in no_pub]
+    return check
+
+
+@check_function()
+def expset_opfsets_unique_titles(connection, **kwargs):
+    '''
+    checks experiment sets with other_processed_files to see if each collection
+    of other_processed_files has a unique title within that experiment set
+    '''
+    check = init_check_res(connection, 'expset_opfsets_unique_titles')
+
+    opf_expsets = ff_utils.search_metadata('search/?type=ExperimentSet&other_processed_files.files.uuid%21=No+value',
```

Comment:
For this and the next check, `page_limit` on the search could be set to 20 or something else low